### DISCUSSION TITLE
opt_update

### DISCUSSION THREAD

**Discussion by pharringtonp19** [google-deepmind/optax#147] [Q&A] [answered]
Just out of curiosity, when would we need to include `params` in `opt_update(grad, opt_state, params)`? 

-- I just realized that fromage requires it

**Comment by mtthss** [comment, marked as answer]
There is no downside to just always passing the `params`,
the transformations that don't need it will just ignore it.

If instead you don't pass it then it will still work fine for transformations that don't need the `params`,
but certain transforms that require them will throw an error to let you know that this is the case.
